### PR TITLE
[2.19.x Backport] [GEOS-10336]-INSPIRE failure: version not propagated in GetCapabilities LegendURL

### PR DESCRIPTION
**Pull request by yashwanthanands**
[GEOS-10336](https://osgeo-org.atlassian.net/browse/GEOS-10336)

* [GEOS-10336]-INSPIRE failure: version not propagated in GetCapabilities LegendURL

Co-authored-by: Mark Prins <1165786+mprins@users.noreply.github.com>

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by aaime** [comment]
@yashwanthanands these two backport PRs are failing on the docs because of the Sphink link issue discussed on geoserver-devel. 

Could you please backport commit https://github.com/geoserver/geoserver/commit/a55fbe513d6f0f75f5643d4c5b729859b05afd74 from main? Open two new PRs for it, one against 2.19.x, one against 2.20.x. Once those are merged, you can rebase these two INSPIRE related and we can move on with a green build.

**Comment by aaime** [comment]
The PR fixing docs has been merged, can you rebase this one?

File: src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesReponseTest.java [hunk 1]
```diff
@@ -35,4 +35,5 @@ public void testSimple() throws Exception {
                 "inline; filename=getcapabilities_1.3.0.xml",
                 result.getHeader("content-disposition"));
     }
+
```
Comment (bradh) [inline src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesReponseTest.java:38]: Not sure why this change, but its causing the QA checks to fail:

```
Error: 0,836 [ERROR] Found 1 non-complying files, failing build
Error: 0,836 [ERROR] To fix formatting errors, run "mvn fmt:format"
Error: 0,836 [ERROR] Non complying file: /home/runner/work/geoserver/geoserver/src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesReponseTest.java
```
